### ISSUE TITLE
obfuscated_if_else should not trigger when rustfmt would change one line into five

### ISSUE THREAD

**Issue by rsalmei** [rust-lang/rust-clippy#13196]
### Description

Although this [lint](https://rust-lang.github.io/rust-clippy/master/index.html#/obfuscated_if_else) is great, it's leaving a sour taste in my mouth for a while.
I occasionally use `.then_some(..).unwrap_or(..)` for a prettier one-line code. And this lint triggers, only to let rustfmt change my pretty one-line code into a five-line one.

Example:

```rust
// original.
let rm = args().shallow.then_some(Shallow).unwrap_or(Recurse(kind));
```

```rust
// once I apply the suggested fix.
let rm = if args().shallow { Shallow } else { Recurse(kind) };
```

```rust
// and then, once I apply rustfmt...
let rm = if args().shallow {
    Shallow
} else {
    Recurse(kind)
};
```

I believe this should only trigger when the resulting code can also be formatted on a single line; otherwise, it should be avoided. Could this be made more conservative, and avoid triggering in this case?
Example playground: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d71a445060e755b6fcbf67a2905f2455

**Comment by Jarcho** [comment]
rustfmt has [`single_line_if_else_max_width`](https://rust-lang.github.io/rustfmt/?version=v1.6.0&search=#single_line_if_else_max_width) to control when it switches. Though we do sometimes add workarounds for what rustfmt produces, this is going to be out of scope since it requires knowing what rustfmt would do in every single case.

**Comment by rsalmei** [comment]
I prefer to avoid any changes to the rustfmt configuration, so I always stick with the default options.
Is it possible to adapt it for this specific situation?

**Comment by Jarcho** [comment]
I've nominated this for the next team meeting (on 2024-08-06). I'm guessing this will end up on a no given that this is, in general, dependent on both rustfmt's configuration and on the surrounding code.

**Comment by rsalmei** [comment]
I understand, thanks @flip1995.
It's just that I feel cheated by this example:
<img width="288" alt="image" src="https://github.com/user-attachments/assets/350ad327-1b24-4f66-a3e7-d46bf4c382e5">

It implies that the changed code will continue to use a single line...

Perhaps if the example was something that breaks, it would convey the change better:
```
Example

func().then_some("then case").unwrap_or("else case");

Use instead:

if func() {
    "then case"
} else {
    "else case"
};
```

**Comment by flip1995** [comment]
That might be a good idea. We're open to doc improvement contributions ;)
